src/bootstrap/lib.rs

```diff
@@ -1135,10 +1135,10 @@ impl Build {
             let krate = &self.crates[&krate];
             if krate.is_local(self) {
                 ret.push(krate);
-                for dep in &krate.deps {
-                    if visited.insert(dep) && dep != "build_helper" {
-                        list.push(*dep);
-                    }
+            }
+            for dep in &krate.deps {
```

Comment:
Why was this change necessary? This function isn't doing in-tree crates anymore? Or is this because we have out of tree crates depending on in-tree crates now? I think this can use a comment.

Reply:
Oh this wasn't necessary per se but it was a preemptive strike because crates from crates.io may now depend on path dependencies via `[patch]`